src/binary-operator-printers/comparison.js

```diff
@@ -23,14 +23,16 @@ module.exports = {
     const groupIfNecessary = groupIfNecessaryBuilder(path);
     const indentIfNecessary = indentIfNecessaryBuilder(path);
 
+    const right = concat([node.operator, line, path.call(print, 'right')]);
+    const parent = path.getParentNode();
+    const shouldGroup =
+      parent.type !== node.type && node.left.type !== node.type;
```

Comment:
Could you also document why this particular type mismatch deserves a grouping?

Reply:
:+1: Will do!